src/components/maps/CameraViewer.jsx

```diff
@@ -0,0 +1,188 @@
+          <Box className="camera-metadata">
+            {this.renderMetadata()}
+          </Box>
+          <Box className="camera-data">
+            {this.renderDataPaging()}
+            {this.renderData()}
+          </Box>
+          <Button label="Done" onClick={()=>{ Actions.mapexplorer.resetActiveCamera() }}></Button>
+        </Box>
+      </Layer>
+    );
+  }
+
+  renderMetadata() {
+    if (this.props.activeCameraMetadataStatus === MAPEXPLORER_CAMERA_STATUS.FETCHING) {
+      return <SpinningIcon />;
+    } else if (this.props.activeCameraMetadataStatus === MAPEXPLORER_CAMERA_STATUS.ERROR) {
+      return <Notification message='ERROR' status='critical' />
+    }
+    
+    if (this.props.activeCameraMetadataStatus === MAPEXPLORER_CAMERA_STATUS.SUCCESS &&
+        this.props.activeCameraMetadata) {
+
+      return (
+        <List>
+        {Object.keys(this.props.activeCameraMetadata).map(key => (
+          <ListItem
+            key={`camera-metadata-${key}`}
+            justify='between'
+            separator='horizontal'
+          >
+            <span>{key}</span>
+            <span>{this.props.activeCameraMetadata[key]}</span>
+          </ListItem>
+        ))}
+        </List>
+      );
+    }
+
+    return null;
+  }
+
+  renderData() {
+    if (this.props.activeCameraDataStatus === MAPEXPLORER_CAMERA_STATUS.FETCHING) {
+      return <SpinningIcon />;
+    } else if (this.props.activeCameraDataStatus === MAPEXPLORER_CAMERA_STATUS.ERROR) {
+      return <Notification message='ERROR' status='critical' />
+    }
+    
+    if (this.props.activeCameraDataStatus === MAPEXPLORER_CAMERA_STATUS.SUCCESS &&
+        this.props.activeCameraData) {
+      const data = this.props.activeCameraData;
+      const items = [];
+      for (let i = this.state.page * ITEMS_PER_PAGE;
+           i < data.length && i < (this.state.page + 1) * ITEMS_PER_PAGE;
+           i++) {
+        let location = data[i].location || '';
+        
+        //Thumbnails!
+        //----------------
+        if (/zooniverse\.org/i.test(location)) {
+          location =
+            `${THUMBNAIL_SERVER_URL}${THUMBNAIL_WIDTH}x${THUMBNAIL_HEIGHT}/` +
+            location.replace(/^(https?|ftp):\/\//, '');
+        }
+        //----------------
+        
+        
+        items.push(
+          <Tile className="camera-data-item" key={`camera-data-${i}`}>
+            <Anchor href={data[i].location}>
+              <img src={location} />
+            </Anchor>
+          </Tile>
+        );
+      }
+
+      return (
+        <Tiles fill={true} flush={true}>
+          {items}
+        </Tiles>
+      );
+    }
+
+    return null;
+  }
+
+  renderDataPaging() {
+    if (this.props.activeCameraDataStatus === MAPEXPLORER_CAMERA_STATUS.SUCCESS &&
+        this.props.activeCameraData) {
+      return (
+        <Box className="camera-data-paging" direction="row" pad="none" alignSelf="stretch" justify="center" separator="horizontal">
+          <Button
```

Comment:
Noticed this after I submitted the review, but the back button doesn't seem to work for me.

Reply:
NANDEYO? Thanks for the catch, I'll look into this.

To clarify, you mean the `<` previous arrow key doesn't work on Pages 2, 3, 4, etc, correct? I'll dive in.

Reply:
Yeah, button to go to the previous page. I wasn't seeing any errors thrown in console, so it's probably an issue with the math in the handler.